### PR TITLE
feat: Allow field policies to hide private fields

### PR DESCRIPTION
**Pull request by torepettersen**
Closes https://github.com/ash-project/ash/issues/1287

# Contributor checklist

- [ ] Bug fixes include regression tests
- [ ] Chores
- [ ] Documentation changes
- [x] Features include unit/acceptance tests
- [ ] Refactoring
- [ ] Update dependencies

### CODE REVIEW

**Comment by zachdaniel** [comment]
So, looking at the implementation here, this is basically spitting distance from what you actually want, which is something like:

```elixir
field_policies do
  private_fields :include | :hide | :prevent
end
```

So `:include` would allow you to write field policies on private fields, `:hide` would hide them automatically and prevent you from writing field policies about them, and `:prevent` would leave them shown and prevent you from writing policies about them at all.

I'm still personally thinking that private fields ought not to have field policies, but it seems like something that we don't necessarily *need* to stop people from doing.

Would you want to take a stab at enhancing this PR to have those three options instead of a boolean flag?

**Comment by torepettersen** [comment]
Thank you, that is a nice suggestion :blush: I will give it a try.

**Comment by zachdaniel** [comment]
Maybe `:include | :hide | :show` would be better naming.

File: lib/ash/policy/authorizer/authorizer.ex [hunk 1]
```diff
@@ -356,6 +356,20 @@ defmodule Ash.Policy.Authorizer do
...
+    schema: [
+      private_fields: [
+        type: {:one_of, [:show, :hide, :include]},
+        default: Application.compile_env(:ash, :policies)[:private_fields] || :show,
+        doc: """
+        Option for how private fields should be handeled by field policies in internal functions.
+
+        - :show will always show private fields
+        - :hide will always hide private fields
+        - :include will let you to write field policies for private fields and private fields
```
Comment (zachdaniel) [inline lib/ash/policy/authorizer/authorizer.ex:369]: We will want to have this as a single line, and tell users to consult the policies guide for more. Then, put this information there. Large option documentation blocks doesn't look good in the generated docs.

Other than that, this LGTM 🥳

**Review by zachdaniel — APPROVED** [review]

**Comment by zachdaniel** [comment]
This looks great! I will merge this after the next release is cut.

**Comment by zachdaniel** [comment]
🚀 Thank you for your contribution! 🚀

**Pull request merged**